### PR TITLE
feat(community): add `bitsocial community export` command

### LINKED ISSUE #64: Add `bitsocial community export` command (pkc-js 0.0.41 export API)
## Summary

Upgrade `@pkcprotocol/pkc-js` from 0.0.40 to 0.0.41 and expose the new community export API (pkcprotocol/pkc-js#100) in the CLI.

### 1. Upgrade pkc-js
- `npm install --save-exact @pkcprotocol/pkc-js@0.0.41`

### 2. New command: `bitsocial community export [ADDRESS]`
Triggers an export on the RPC server, streams progress, downloads the finished SQLite snapshot via `GET /exports/<exportId>`, and verifies its sha256 against the export record.

Flags:
- `ADDRESS` arg + `--name` / `--publicKey` — community lookup (same trio as `community get`)
- `-o, --path` — local destination file (default: `./<address>.sqlite`)
- `--includePrivateKey` — ask the RPC server to include the community signer's private key in the export
- `--force` — overwrite an existing file at the destination
- `--quiet` — suppress progress output (only print the final file path)
- Ctrl+C cancels the in-flight export via `cancelExport()`

### 3. New daemon flag: `--allowPrivateKeyExport`
Server-side policy passed to the pkc-js WS server options, controlling whether RPC clients may request `includePrivateKey` exports.

### 4. Tests
- Export command: happy path, lookup variants, `--includePrivateKey`, overwrite protection / `--force`, sha256 mismatch failure
- Daemon flag plumbing

## Progress
- [ ] Upgrade pkc-js to 0.0.41
- [ ] `community export` command
- [ ] `daemon --allowPrivateKeyExport` flag
- [ ] Tests
- [ ] Build + full test suite pass

### PR DESCRIPTION
**Pull request by Rinse12**
Closes #64

## Summary

Upgrades `@pkcprotocol/pkc-js` 0.0.40 → 0.0.41 and exposes its new community export API (pkcprotocol/pkc-js#100) in the CLI.

### `bitsocial community export [ADDRESS]`

Triggers an export on the RPC server (daemon), streams progress to stderr, downloads the finished SQLite snapshot via `GET /exports/<exportId>`, and verifies its sha256 against the export record before reporting success.

| Flag | Behavior |
|---|---|
| `ADDRESS` / `--name` / `--publicKey` | community lookup, same trio as `community get` |
| `-o, --path` | local destination file (default `./<address>.sqlite`) |
| `--includePrivateKey` | ask the server to include the community signer's private key (off by default; required for a restorable backup that keeps the same address) |
| `--force` | overwrite an existing destination file |
| `--quiet` | suppress stderr progress; stdout gets only the final path |

Implementation details:
- Downloads to `<dest>.partial` and atomically renames after the sha256 check, so a corrupted/interrupted download never clobbers the destination.
- Ctrl+C aborts via `AbortSignal` → `cancelExport()` on the server; a second Ctrl+C force-exits.
- Errors cleanly when the community is not local to the RPC server (remote communities can't be exported).

### `bitsocial daemon --allowPrivateKeyExport`

Negatable boolean (default `true`, matching the pkc-js default), plumbed through `startDaemonServer` → `PKCWsServer`. With `--no-allowPrivateKeyExport`, RPC clients requesting `--includePrivateKey` get `ERR_PRIVATE_KEY_EXPORT_NOT_ALLOWED`.

## Tests

11 new tests in `test/cli/export.community.test.ts`:
- happy path with a real HTTP server download + sha256 verification + stdout path
- lookup variants (`--name`, `--publicKey`), missing-identifier error
- `--includePrivateKey` pass-through to `community.export()`
- overwrite protection without `--force` (existing file untouched, export never starts) and overwrite with `--force`
- sha256 mismatch fails and leaves neither the destination nor a `.partial` behind
- export error record (`ERR_PRIVATE_KEY_EXPORT_NOT_ALLOWED`) fails the command
- non-local community error
- daemon flag definition (negatable, default true)

Full `npm run test:cli` suite: 220 passed, 1 skipped (pre-existing), 29 files.

## Summary by CodeRabbit

* **New Features**
  * Export community snapshots locally with SHA-256 verification, real-time progress output, Ctrl+C cancellation, and force-overwrite; lookup by address, name, or public key
  * Daemon flag to control private-key export permissions (enabled by default; disable with --no-allowPrivateKeyExport)

* **Tests**
  * Comprehensive test coverage for the export workflow, edge cases, verification, cancellation, and daemon flag behavior

* **Chores**
  * Bumped `@pkcprotocol/pkc-js` to 0.0.41

### CODE REVIEW

File: src/cli/commands/community/export.ts [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+                    if (percent !== lastPrintedPercent) {
+                        lastPrintedPercent = percent;
+                        this._printProgress(quiet, `\rExporting ${community.address}: ${percent}%`);
+                    }
+                }
+            };
+            community.on("exportschange", checkRecords);
+            // The terminal notification may have arrived before the listener was attached
+            checkRecords(community.exports);
+        });
```
[the remark this answers is not in the document]
Comment (Rinse12) [reply src/cli/commands/community/export.ts:202]: Fixed in 779a85f — `_runExport()` now registers an `abort` listener that cleans up and rejects with "Export cancelled" immediately, instead of waiting for the server's terminal record (which never arrives if the daemon died). Note pkc-js's own abort listener is registered first inside `community.export()`, so `cancelExport()` is still dispatched to the server before we reject. Added a regression test (aborted export with a daemon that never emits a terminal record rejects instead of hanging, with listener-cleanup assertion).

File: src/cli/commands/community/export.ts [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    }
+
+    /** Download the finished snapshot to destPath, verifying its sha256 against the export record. */
+    private async _downloadAndVerify(record: ExportRecord, destPath: string, signal: AbortSignal, quiet: boolean) {
+        this._printProgress(quiet, `Downloading snapshot from ${record.url}\n`);
+        const response = await fetch(record.url!, { signal });
+        if (!response.ok || !response.body) {
+            this.error(`Failed to download export from ${record.url}: HTTP ${response.status}`);
```
[the remark this answers is not in the document]
Comment (Rinse12) [reply src/cli/commands/community/export.ts]: Fixed in 779a85f — the record's download URL is now validated before any request is made: its origin must match the RPC server's http origin (ws[s]:// → http[s]://, mirroring pkc-js's `rpcHttpOrigin`) and its path must start with `/exports/`. Worth noting the RPC server is already fully trusted in this threat model (it holds the communities and signer keys), so this is defense-in-depth rather than a vulnerability fix — but it's cheap and enforces the protocol contract. Added a test asserting a wrong-origin URL is refused with zero requests hitting the server.

**Comment by Rinse12** [comment]
Two new commits since the CodeRabbit review:

**52c5ac1 feat(community): default export destination to `<dataPath>/exports/<address>_<datetime>.sqlite`**
- Default `-o` was `./<address>.sqlite` (cwd-relative); snapshots now land in `<dataPath>/exports/` with a datetime suffix (ISO 8601, `:` → `-`, same convention as daemon log filenames) so repeated exports never collide. `-o` still overrides.
- Also closes the test coverage gaps found in review — export tests went from 11 to 28 (HTTP errors, mid-stream drops, URL validation halves, wss→https origin mapping, missing sha256/url, default path, destroy-on-failure, `_runExport` race/abort unit tests, non-quiet stdout/stderr contract).

**0e40b4f fix(daemon): stop express's catch-all 404 from clobbering /exports downloads**
- Found by running a live export against a real daemon: the download always 404'd because express and pkc-js's `/exports/<id>` handler share one http.Server, and express's synchronous 404 beat the async pkc-js handler. Fixed with a middleware that stays silent for `/exports/<uuid>` (deliberately unmounted — `app.use("/exports", …)` mutates the shared `req.url` and turns the 404 into a hang).
- Regression test verified to fail (404) against the unfixed build; plumbing tests assert `--allowPrivateKeyExport` reaches `PKCWsServer` verbatim.

Verified live end-to-end (export → download → sha256 verify → `<dataPath>/exports/…_2026-06-07T05-03-35.773Z.sqlite`, valid SQLite db, exit 0). Full suite: 241 passed, 1 skipped, 29 files.

Analysis details in #64.
